docs/report_stage_i.md

```diff
@@ -0,0 +1,497 @@
+    4. Allow to do "measure total" instrumentation fairly and with full coverage of execution
+    5. (opt) Allow to do "measure batch" instrumentation
+    5. (opt) Possibly adjust the instrumentation to doing the modelling of duration via the binomial model
+    6. (opt) Try and compare alternative wallclock timers (e.g. PAPI)
+    7. (opt) Provide full support for CPU cycles using TSC libraries
+    8. Write down the instrumentation and measurement standard ruleset
+    9. (opt) Try further lowering the overhead of wallclock timers
+    10. Provide consistent treatment of warm-up
+    11. Analyze warm-up conditions versus normal EVM/eWASM operation within an Ethereum node
+    12. Support gathering of timer overhead measuremnts during OPCODE measurements
+    13. Support for instrumentation in at least one another eWASM implementation
+    14. Rethink the approach to garbage collection for environments that have one
+3. Analysis tasks
+    1. Tidy and improve structure of the existing analysis code in R, allow for its extensibility
+    1. Improve the visualization method to check distributions of measurements per OPCODE and environment (frequency plots used in Stage I are not useful)
+    1. Look into the extreme outliers in OPCODE measurements, if they are still present after rectifying the instrumentation stack
+    1. Revisit the choice of the pivot OPCODE for implementation-relative measurements, describe criteria for this choice
+    1. Allow analysis of OPCODE measurements coming from programs in which their execution has been parametrized (via arguments, context, surroundings)
+    2. Explore validation methods proposed
+    3. Implement automated scripts for the chosen validation methods
+    4. (opt) Implement (or adapt an existing implementation or existing data sets, if they exist) validation via blockchain history
+    5. Give recommendations for any gas cost adjustments which could be safely done at current stage (i.e. ones which are reasonable consistently between all environments measured)
+    6. Identify OPCODEs which will not have a gas cost suitable for all environments and try to find the cause and recommend optimizations
+    6. Answer "Q1: Can we devise a one-size-fits-all set of OPCODE gas costs?"
+    7. Provide a quantitative argument (e.g. a statistical test) that for every OPCODE, measurement of the timer overhead is significantly smaller
+    7. Answer "Q2: is measuring of individual instructions feasible?"
+    8. (opt) Devise scoring functions which can be used to adaptively generate programs
+    8. (opt) Provide a performant implementations of the scoring functions which can be reasonably plugged into an search algorithm for adaptive program generation
+    8. (opt) Answer "Q3: How can we explore the entire program space to capture all sources of variability of OPCODEs computational cost"
+    9. Answer "Q4: How should warm-up be treated?"
```

Comment:
Instrumentation warmup vs actual contract startup overhead might be discussed here.